src/ydata_synthetic/preprocessing/regular/data_processor.py

```diff
@@ -0,0 +1,79 @@
+from typing import List
+
+from numpy import concatenate, ndarray, split, zeros
+from pandas import concat, DataFrame
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import MinMaxScaler, OneHotEncoder
+from typeguard import typechecked
+
+
+class DataProcessor(BaseEstimator, TransformerMixin):
+    """
+    Main class for Data Preprocessing. It is a base version.
+    It works like any other transformer in scikit lear with the methods fit, transform and inverse transform.
+    Args:
+        num_cols (list of strings):
+            List of names of numerical columns
+        cat_cols (list of strings):
+            List of names of categorical columns
+    """
+    @typechecked
```

Comment:
You can perform the _typechecked_ decorator at the level of the class and only use the _typeguard_ignore_ decorator in the methods that you don't want to get checked based on the annotations.

Reply:
Thanks, nice tip, makes it more readable